float32 F_area1
float32 x_area1
float32 y_area1
float32 F_area2
float32 x_area2
float32 y_area2
float32 F_area3
float32 x_area3
float32 y_area3
float32 F_net
float32 x_net
float32 y_net
int8 contact
